pkg/neorpc/types.go

```diff
@@ -115,6 +115,15 @@ func (s *SignerWithWitness) UnmarshalJSON(data []byte) error {
 	if err != nil {
 		return fmt.Errorf("not a signer: %w", err)
 	}
+	if len(aux.AllowedContracts) > transaction.MaxAttributes {
+		return fmt.Errorf("invalid number of AllowedContracts: got %d, allowed %d at max", len(aux.AllowedContracts), transaction.MaxAttributes)
+	}
+	if len(aux.AllowedGroups) > transaction.MaxAttributes {
+		return fmt.Errorf("invalid number of AllowedGroups: got %d, allowed %d at max", len(aux.AllowedGroups), transaction.MaxAttributes)
+	}
+	if len(aux.Rules) > transaction.MaxAttributes {
+		return fmt.Errorf("invalid number of Rules: got %d, allowed %d at max", len(aux.Rules), transaction.MaxAttributes)
+	}
```

Comment:
Ref. https://github.com/nspcc-dev/neo-go/issues/3226. But this restriction is not only for DOS preventing, but also to forbid invalid transactions unmarshalling. Because if transaction has more than 16 Rules/Contracts/Groups then it's invalid (and this check is present in the binary deserializer). So I consider we should keep it.